### PR TITLE
Update `Suite2pSegmentationInterface` to support multiple channels and planes

### PR DESCRIPTION
**Pull request by weiglszonja**
- [x] Update Suite2pSegmentationExtractor to support multiple channels and planes https://github.com/catalystneuro/roiextractors/pull/242
- [x] Update all necessary metadata (Created #603 to discuss these questions)
- [x] Add test for stubbing, as mentioned in https://github.com/NeurodataWithoutBorders/nwb-guide/pull/460#issuecomment-1775175196
- [x] Update docs for usage example

- [x] Remove pin for roiextractors

### CODE REVIEW

File: src/neuroconv/datainterfaces/ophys/suite2p/suite2pdatainterface.py [hunk 1]
```diff
@@ -1,19 +1,74 @@
...
+            return metadata
+
+        plane_name_suffix = f"{self.channel_name.capitalize()}{self.plane_name.capitalize()}"
+        imaging_plane_metadata = metadata["Ophys"]["ImagingPlane"][0]
+        imaging_plane_name = imaging_plane_metadata["name"] + plane_name_suffix
+        imaging_plane_metadata.update(name=imaging_plane_name)
+        plane_segmentation_metadata = metadata["Ophys"]["ImageSegmentation"]["plane_segmentations"][0]
+        plane_segmentation_metadata.update(
+            name=f"PlaneSegmentation{plane_name_suffix}",
+            # imaging_plane=imaging_plane_name,
```
Comment (weiglszonja) [inline src/neuroconv/datainterfaces/ophys/suite2p/suite2pdatainterface.py:63]: will uncomment once #594 is merged to add the link to the same imaging plane

**Comment by weiglszonja** [comment]
Noticed an issue with the extractor, as it overrides the `channel_name` to "OpticalChannel" or capitalised version of the original channel name (chan1 or chan2) to use it for the name of the OpticalChannel it will be created, the override results in incorrect read for the mean image, fixed in:
https://github.com/catalystneuro/roiextractors/pull/263

File: src/neuroconv/datainterfaces/ophys/requirements.txt [hunk 2]
```diff
@@ -1 +1 @@
-roiextractors>=0.5.3
+roiextractors @ git+https://github.com/catalystneuro/roiextractors.git@main
```
Comment (weiglszonja) [inline src/neuroconv/datainterfaces/ophys/requirements.txt:1]: @CodyCBakerPhD 
I'm leaving this PR in draft until a new version is cut for `roiextractors`, just to keep in mind this is ready

**Comment by weiglszonja** [comment]
@CodyCBakerPhD on my end it is, I just listed @alessandratrapani for review

**Review by alessandratrapani — APPROVED** [review]
It looks all right to me

**Pull request merged**